quarkchain/cluster/shard_state.py

```diff
@@ -1183,26 +1183,32 @@ def get_next_block_coinbase_amount(self):
         # TODO: add x-shard tx
         return coinbase
 
-    def __get_all_unconfirmed_header_list(self) -> List[MinorBlockHeader]:
+    def __get_unconfirmed_header_list(self, max_blocks) -> List[MinorBlockHeader]:
         """height in ascending order"""
         header_list = []
         header = self.header_tip
         start_height = (
             self.confirmed_header_tip.height if self.confirmed_header_tip else -1
         )
-        for i in range(header.height - start_height):
-            header_list.append(header)
+        steps = header.height - start_height
+        Logger.debug("{} blocks to be confirm".format(steps))
+        max_height = start_height + max_blocks
+        st = time_ms()
+        for i in range(steps):
+            if header.height <= max_height:
+                header_list.append(header)
             header = self.db.get_minor_block_header_by_hash(
                 header.hash_prev_minor_block
             )
+        Logger.info("get_unconfirmed_header_list: unconfirmed header {}, max_blocks = {}, time use {} ms."
```

Comment:
What is the purpose of logging info here?

Reply:
From this log you can see:
1. Whether there is Header accumulation;
2. If there is accumulation, how much does it affect the performance;
3. Is there any message congestion? (When multiple pieces of this message from the same shard appear at once but have not appeared for some times before, may this message be blocked?)